plugins/plugin-github/che-plugin-github-factory-resolver/src/main/java/org/eclipse/che/plugin/github/factory/resolver/GithubFactoryParametersResolver.java

```diff
@@ -66,30 +81,35 @@ public boolean accept(@NotNull final Map<String, String> factoryParameters) {
    */
   @Override
   public FactoryDto createFactory(@NotNull final Map<String, String> factoryParameters)
-      throws BadRequestException {
+      throws BadRequestException, ServerException {
 
     // no need to check null value of url parameter as accept() method has performed the check
-    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.get("url"));
+    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.get(URL_PARAMETER_NAME));
 
     // create factory from the following location if location exists, else create default factory
-    FactoryDto factory = urlFactoryBuilder.createFactory(githubUrl.factoryJsonFileLocation());
-
+    FactoryDto factory =
+        urlFactoryBuilder
+            .createFactoryFromDevfile(githubUrl.devfileFileLocation())
+            .orElseGet(
```

Comment:
IMHO Hard to say that using double `orElseGet` makes code clearer.
Up to you revert it to if/else statements or use lambda expressions.

Reply:
if else with optionals and variable re-assigning looks messy too.this way looks not very complex for me